prime-dotnet-webapi/Controllers/EnrolleePaperSubmissionsController.cs

```diff
@@ -243,20 +254,111 @@ public async Task<ActionResult> GetAdjudicationDocuments(int enrolleeId)
+        /// </summary>
+        /// <param name="dateOfBirth"></param>
+        [HttpHead("paper-submissions", Name = nameof(CheckForPotentialPaperEnrolleeReturnee))]
+        [Authorize(Roles = Roles.PrimeEnrollee)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult> CheckForPotentialPaperEnrolleeReturnee([FromQuery] DateTime dateOfBirth)
+        {
+            var result = await _enrolleePaperSubmissionService.PotentialReturneeExistsAsync(dateOfBirth);
+
+            if (!result)
+            {
+                return NotFound();
+            }
+
+            return Ok();
+        }
+
+        // POST: api/Enrollees/5/potential-paper-enrollee
+        /// <summary>
+        /// Creates a new Enrollee who may have a a previous paper enrolment.
+        /// </summary>
+        [HttpPost("{enrolleeId}/potential-paper-enrollee", Name = nameof(CreateLinkWithPotentialPaperEnrollee))]
+        [Authorize(Roles = Roles.TriageEnrollee + "," + Roles.PrimeEnrollee)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult> CreateLinkWithPotentialPaperEnrollee(int enrolleeId, EnrolleeLinkedEnrolmentViewModel payload)
+        {
+            await _enrolleePaperSubmissionService.CreateInitialLinkAsync(enrolleeId, payload.UserProvidedGpid);
+            return Ok();
+        }
+
+        // PUT: api/Enrollees/5/linked-gpid
+        /// <summary>
+        /// Updates the paper enrolment gpid that the user provided
+        /// </summary>
+        [HttpPut("{enrolleeId}/linked-gpid", Name = nameof(UpdateGpidLinkToPaperEnrollee))]
+        [Authorize(Roles = Roles.TriageEnrollee + "," + Roles.PrimeEnrollee)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult> UpdateGpidLinkToPaperEnrollee(int enrolleeId, EnrolleeLinkedEnrolmentViewModel payload)
+        {
+            var record = await _enrolleeService.GetPermissionsRecordAsync(enrolleeId);
+            if (record == null)
+            {
+                return NotFound($"Enrollee not found with id {enrolleeId}");
+            }
+            if (!record.MatchesUserIdOf(User))
+            {
+                return Forbid();
+            }
+
+            await _enrolleePaperSubmissionService.UpdateLinkedGpidAsync(enrolleeId, payload.UserProvidedGpid);
```

Comment:
Should we prevent an Enrollee for updating a link if it's confirmed?

Reply:
A confirmed link only happens after submission and passing the rules engine, once confirmed that paper enrolment will not be part of the check, so it will not show up again even if the same enrollee logs in. The way it's set up now will prevent even if 2 people have a potential link, but the link is confirmed with the second one, then that page won't show up if the other person log in after the first person submits. At the end, if a link is confirmed the enrollee will not be able to update it.